libbinlogevents/src/gtids/gtidset.cpp

```diff
@@ -235,7 +236,10 @@ bool Gtid_set::is_empty() const { return m_gtid_set.empty(); }
 std::size_t Gtid_set::count() const {
   std::size_t count{0};
 
+  MY_COMPILER_DIAGNOSTIC_PUSH()
+  MY_COMPILER_GCC_DIAGNOSTIC_IGNORE("-Wunused-variable")
   for (auto const &[uuid, intervals] : m_gtid_set) {
```

Comment:
Would `for([[maybe_unused]] auto const ...)` work?

Reply:
@dutow , I tried but gcc 7 has no implemented `[[maybe_unused]]` for structure bindings.